static/js/operation.js

```diff
@@ -135,10 +139,159 @@ function parseTable(text, data) {
+ */
+function parseAssembly(text, data) {
+  var lines = splitLines(text);
+  if (lines.length === 1) throw 'Error: there is only one line.';
+
+  var format = getFileFormat(text); // read file format of the inputted text
+  var types = ['id'];
+  var cols = ['id', 'length', 'gc', 'coverage']; // the information available in the contig titles
+  var df = [];
+
+  var id = 0;
+  var length = 0;
+  var gc = 0;
+  var coverage = 0;
+  var contig_length = 0;
+  var GC_count = 0;
+
+  for (var i = 0; i < lines.length; i++) {
+    var line = lines[i];
+    if (line.toUpperCase().match(/^[ATGCU]+$/g)) { // Checking if the current line is not a contig title
+      [GC_count, contig_length] = calculateGCAndLength(line, GC_count, contig_length);
+      gc = ((GC_count/contig_length).toFixed(2)).toString(); // idnetified last component of contig title
+    }
+    else {
+      if (i !== 0) {
+        if (length <= 1000) {
+          continue;
+        }
+        df.push([id, length, gc, coverage]) // append dataframe with contig title information
+        GC_count = 0;
+        contig_length = 0;
+      }
+      [id, length, coverage] = parseContigTitles(line, format);
+      coverage = (Number(coverage).toFixed(3)).toString();
+    }
+  }
+  if (df.length === 0) throw 'Error: No contig is bigger than 1000 base pairs.';
+
+
+  var deci = {};
+  var cats = {};
+  var feats = {};
+
+  // identify field types and re-format data
+  var types = ['id'];
+  for (var i = 1; i < cols.length; i++) {
+    var arr = [];
+    for (var j = 0; j < df.length; j++) {
+      arr.push(df[df.length - j - 1][i]);
+    }
+      var x = parseFieldType(cols[i], arr);
+      var type = x[0];
+      var col = x[1];
+      types.push(type); // identified type
+      cols[i] = col; // updated name
+      for (var j = 0; j < df.length; j++) {
+        df[j][i] = arr[j];
+      }
+  }
+    // summarize categories or features
+    switch (type) {
+      case 'number':
+        deci[col] = maxDecimals(arr);
+        break;
+      case 'category':
+        cats[col] = listCats(arr);
+        break;
+      case 'feature':
+        feats[col] = listFeats(arr);
+        break;
+    }
 
+  data.cols = cols;
+  data.types = types;
+  data.features = [];
+  data.df = df;
   return [deci, cats, feats];
 }
 
+/**
+ * Calculates GC count in and length of one line in the contig.
+ * @function calculateGCAndLength
+ * @param {String} line - one line string in contig
+ * @param {Integer} numGC - current counter of GC instances in contig
+ * @param {Integer} totalBP - current counter of length of contig
+ * @returns {Array.<Integer, Integer>} - updated GC count of contig, updated length of contig
+ */
+function calculateGCAndLength(line, numGC, totalBP) {
+  var count = numGC;
+  var length = totalBP + line.length;
+  // iterating through the line to find 'G' or 'C'
+  for(var i = 0; i < line.length; i++) {
+      if (line.charAt(i).toUpperCase() == 'G' || line.charAt(i).toUpperCase() == 'C') {
```

Comment:
I am thinking if we can adopt the [IUPAC codes](https://www.bioinformatics.org/sms/iupac.html) (also mentioned above) for account for nucleobase ambiguity. For example, if the character is "R" (which means either "A" or "G"), then we should count GC and 1/2 times. If the character is "N", we should count it as 1/4 times. To achieve this, a convenient JavaScript structure is the `switch ... case` statement. For example:

```js
switch (line.charAt(i).toUpperCase()) {
  case 'G':
  case 'C':
  case 'S':
    count ++;
    break;
  case 'R':
  case 'Y':
  case 'K':
  case 'M':
    count += 0.5;
    break;
  ...
}
```